### PR TITLE
fs/fs_sync/ioctl: update 

### PR DESCRIPTION
**Pull request by Donny9**
## Summary

fs/fs_sync: support file sync operation by ioctl with BIOC_FLUSH

fs/fs_ioctl:  support get sector size by cmd BIOC_BLKSSZGET

## Impact
support BIOC_BLKSSZGET
## Testing
Vela CI

### CODE REVIEW

File: fs/vfs/fs_ioctl.c [hunk 1]
```diff
@@ -110,6 +110,18 @@ int file_vioctl(FAR struct file *filep, int req, va_list ap)
             ret = inode_getpath(inode, (FAR char *)(uintptr_t)arg);
           }
         break;
+
+      case BIOC_BLKSSZGET:
+        if (inode->u.i_ops != NULL && inode->u.i_ops->ioctl != NULL)
+          {
+            struct geometry geo;
+            ret = inode->u.i_ops->ioctl(filep, BIOC_GEOMETRY,
+                                        (unsigned long)&geo);
```
Comment (pkarashchenko) [inline fs/vfs/fs_ioctl.c:119]: Do we need
```suggestion
                                        (unsigned long)(uintptr_t)&geo);
```
?

Comment (Donny9) [reply fs/vfs/fs_ioctl.c]: Done, thank you.

File: fs/vfs/fs_fsync.c [hunk 2]
```diff
@@ -53,23 +54,31 @@
...
    * the mountpoint operations vtable contains a sync method.
    */
 
   inode = filep->f_inode;
-  if (!inode || !INODE_IS_MOUNTPT(inode) ||
-      !inode->u.i_mops || !inode->u.i_mops->sync)
+  if (inode != NULL)
     {
-      return -EINVAL;
+      if (INODE_IS_MOUNTPT(inode) && inode->u.i_mops->sync)
```
Comment (hartmannathan) [inline fs/vfs/fs_fsync.c:68]: Before checking `inode->u.i_mops->sync` I think we need to check `inode->u.i_mops` is not NULL, else sync may be in a NULL or undefined location.

File: fs/vfs/fs_fsync.c [hunk 3]
```diff
@@ -53,23 +54,31 @@
...
+        {
+          /* Yes, then tell the mountpoint to sync this file */
+
+          return inode->u.i_mops->sync(filep);
+        }
+      else if (inode->u.i_ops->ioctl)
```
Comment (hartmannathan) [inline fs/vfs/fs_fsync.c:74]: Before checking `inode->u.i_ops->ioctl` I think we need to check `inode->u.i_ops` is not NULL, for same reason as above.

**Review by hartmannathan — CHANGES_REQUESTED** [review]

File: fs/vfs/fs_fsync.c (hunk 2 above)
Comment (Donny9) [reply fs/vfs/fs_fsync.c]: Done! Thank you.

File: fs/vfs/fs_fsync.c (hunk 3 above)
Comment (Donny9) [reply fs/vfs/fs_fsync.c]: inode->u.i_mops

Comment (Donny9) [reply fs/vfs/fs_fsync.c]: Done. Thank you~

File: fs/vfs/fs_fsync.c (hunk 2 above)
Comment (Donny9) [reply fs/vfs/fs_fsync.c]: Done. Thank you~

**Review by hartmannathan — APPROVED** [review]

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by pkarashchenko — APPROVED** [review]

**Pull request merged**